code/game/machinery/medical_kiosk.dm

```diff
@@ -0,0 +1,345 @@
+	var/paying_customer = FALSE		//Ticked yes if passing inuse()
+
+	var/datum/bank_account/account  //payer's account.
+	var/mob/living/carbon/human/H   //The person using the console in each instance. Used for paying for the kiosk.
+	var/mob/living/carbon/human/altPatient   //If scanning someone else, this will be the target.
+	var/obj/item/card/id/C          //the account of the person using the console.
+
+/obj/machinery/medical_kiosk/Initialize() //loaded subtype for mapping use
+	. = ..()
+	scanner_wand = new/obj/item/scanner_wand(src)
+
+/obj/machinery/medical_kiosk/proc/inuse()  //Verifies that the user can use the interface, followed by showing medical information.
+	if (pandemonium == TRUE)
+		active_price += (rand(10,30)) //The wheel of capitalism says health care ain't cheap.
+	if(!istype(C))
+		say("No ID card detected.") // No unidentified crew.
+		return
+	if(C.registered_account)
+		account = C.registered_account
+	else
+		say("No account detected.")  //No homeless crew.
+		return
+	if(account?.account_job?.paycheck_department == payment_department)
+		use_power(20)
+		paying_customer = TRUE
+		say("Hello, esteemed medical staff!")
+		RefreshParts()
+		return
+	if(!account.has_money(active_price))
+		say("You do not possess the funds to purchase this.")  //No jobless crew, either.
+		return
+	else
+		account.adjust_money(-active_price)
+		var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_MED)
+		if(D)
+			D.adjust_money(active_price)
+		use_power(20)
+		paying_customer = TRUE
+	icon_state = "kiosk_active"
+	say("Thank you for your patronage!")
+	RefreshParts()
+	return
+
+/obj/machinery/medical_kiosk/proc/clearScans() //Called it enough times to be it's own proc
+	scan_active_1 = FALSE
+	scan_active_2 = FALSE
+	scan_active_3 = FALSE
+	scan_active_4 = FALSE
+	return
+
+/obj/machinery/medical_kiosk/update_icon_state()
+	if(is_operational())
+		icon_state = "kiosk_off"
+	else
+		icon_state = "kiosk"
+
+/obj/machinery/medical_kiosk/wrench_act(mob/living/user, obj/item/I) //Allows for wrenching/unwrenching the machine.
+	..()
+	default_unfasten_wrench(user, I, time = 10)
+	return TRUE
+
+/obj/machinery/medical_kiosk/RefreshParts()
+	var/obj/item/circuitboard/machine/medical_kiosk/board = circuit
+	if(board)
+		active_price = board.custom_cost
+	return
+
+/obj/machinery/medical_kiosk/attackby(obj/item/O, mob/user, params)
+	if(default_deconstruction_screwdriver(user, "kiosk_open", "kiosk", O))
+		return
+	else if(default_deconstruction_crowbar(O))
+		return
+
+	if(istype(O, /obj/item/scanner_wand))
+		var/obj/item/scanner_wand/W = O
+		if(scanner_wand)
+			to_chat(user, "<span class='warning'>There's already a scanner wand in [src]!</span>")
+			return
+		if(HAS_TRAIT(O, TRAIT_NODROP) || !user.transferItemToLoc(O, src))
+			to_chat(user, "<span class='warning'>[O] is stuck to your hand!</span>")
+			return
+		user.visible_message("<span class='notice'>[user] snaps [O] onto [src]!</span>", \
```

Comment:
What happens if someone walks off with the wand and steals it entirely?

Perhaps you could have the scanner have a sort of "scanner plate" /pressure plate that you walk onto, and then it scans you.

"Please step onto the scanning pad"

This then removes the need to store two patients or whatever.